geopandas/testing.py

```diff
@@ -133,15 +133,49 @@ def assert_geoseries_equal(
     if not check_crs:
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore", "CRS mismatch", UserWarning)
-            if check_less_precise:
-                assert geom_almost_equals(left, right)
-            else:
-                assert geom_equals(left, right)
+            check_equality(left, right, check_less_precise)
     else:
-        if check_less_precise:
-            assert geom_almost_equals(left, right)
-        else:
-            assert geom_equals(left, right)
+        check_equality(left, right, check_less_precise)
+
+
+def check_equality(left, right, check_less_precise):
+    assert_error_message = (
+        "{} out of {} geometries are not equal.\n"
+        "Indeces where geometries are not equal: {} \n"
```

Comment:
```suggestion
        "Indices where geometries are not equal: {} \n"
```

Reply:
Oh sorry for the typo. About comparing the message, a question arises: Should I copy paste the error in this test https://github.com/geopandas/geopandas/blob/d8dfa01cfca4a83f61dae87c8253e5594c5b3264/geopandas/tests/test_testing.py#L66-L67? 

It seems a bad practice to me. Maybe I should have it declared as a constant in testing.py and then import it in test_testing.py?

Reply:
I would not overcomplicate it. Capture the string and check that specific strings are part of it. As `assert "1 out of 2" in err`, `"are not equal: [0]" in err` etc. This does not need to be super fancy, we just want to make sure that the message we return does not change due to change elsewhere in the code.